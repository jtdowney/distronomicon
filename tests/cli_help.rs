use assert_cmd::cargo::cargo_bin_cmd;
use insta::assert_snapshot;

#[test]
fn test_help_output() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}

#[test]
fn test_check_subcommand_help() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("check").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}

#[test]
fn test_update_subcommand_help() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("update").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}

#[test]
fn test_version_subcommand_help() {
    let mut cmd = cargo_bin_cmd!("distronomicon");
    let output = cmd.arg("version").arg("--help").output().unwrap();

    assert!(output.status.success());

    let stdout = String::from_utf8_lossy(&output.stdout);
    assert_snapshot!(stdout);
}
