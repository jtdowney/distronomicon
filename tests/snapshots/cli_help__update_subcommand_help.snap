---
source: tests/cli_help.rs
expression: stdout
---
Update to the latest release

Usage: distronomicon --app <APP> update [OPTIONS] --repo <REPO> --pattern <PATTERN> --state-directory <STATE_DIRECTORY>

Options:
      --repo <REPO>
          GitHub repository (owner/name)
      --pattern <PATTERN>
          Asset filename pattern (regex)
      --state-directory <STATE_DIRECTORY>
          State directory [env: STATE_DIRECTORY=]
      --checksum-pattern <CHECKSUM_PATTERN>
          Checksum filename pattern (optional)
      --github-token <TOKEN>
          GitHub API token [env: GITHUB_TOKEN]
      --github-host <HOST>
          GitHub API host [env: GITHUB_HOST=] [default: api.github.com]
      --allow-prerelease
          Allow prerelease versions
      --restart-command <RESTART_COMMAND>
          Command to run after successful update
      --retain <RETAIN>
          Number of releases to retain [default: 3]
      --skip-verification
          Skip checksum verification
  -h, --help
          Print help
